AdminWebsite/AdminWebsite/ClientApp/src/app/booking/participant/list/participant-list.component.ts

```diff
@@ -37,18 +37,22 @@ export class ParticipantListComponent implements OnInit, OnChanges, DoCheck {
         const judicialMembersLocal =
             this.hearing?.judiciaryParticipants
                 ?.map(j => ({ email: j.email, displayName: j.displayName, role: j.roleCode }))
-                .sort((a, b) => a.displayName.localeCompare(b.displayName)) ?? [];
+                .sort(this.sortByDisplayNameThenByEmail()) ?? [];
         const sortedJudicialMembersLocal =
             this.sortedJudiciaryMembers
                 ?.map(j => ({ email: j.email, displayName: j.display_name, role: j.hearing_role_code }))
-                .sort((a, b) => a.displayName.localeCompare(b.displayName)) ?? [];
+                .sort(this.sortByDisplayNameThenByEmail()) ?? [];
 
         const judiciaryEmailListChanged = JSON.stringify(judicialMembersLocal) !== JSON.stringify(sortedJudicialMembersLocal);
         if (judiciaryEmailListChanged) {
-            this.sortJudiciaryMembers();
+            this.sortJoh();
         }
     }
 
+    sortJoh() {
+        this.sortJudiciaryMembers();
+    }
```

Comment:
Why are wrapping sortJudiciaryMembers() in another function, rather than calling directly?

Reply:
it's purely for the unit tests, so that we can do this:

```
spyOn(component, 'sortJoh').and.callFake(() => {
                sortJudiciaryMembersCalled = true;
                component.sortJudiciaryMembers();
            });
```

I wanted to mock `sortJudiciaryMembers()` so that we can set sortJudiciaryMembersCalled = true when it is called. However we still need to invoke the actual code within `sortJudiciaryMembers()` so that the judiciary members are still sorted.

as you can't spy on a method and have it call itself, I had to separate it into 2 methods, so that we can spy on `sortJoh()` while having it call the actual sorting method (`sortJudiciaryMembers()`).